doc/_templates/autoapi/python/class.rst

```diff
@@ -0,0 +1,56 @@
+{% if obj.display %}
+
+{% if obj.docstring %}
+{{ obj.docstring|prepare_docstring }}
+{% endif %}
+
+.. {{ obj.type }}:: {{ obj.short_name }}{% if obj.args %}({{ obj.args }}){% endif %}
+{% if obj.constructor %}
+
+{% for (args, return_annotation) in obj.constructor.overloads %}
+   {% if args and args.startswith("self, ") %}{% set args = args[6:] %}{% endif %}
+   {{ " " * (obj.type | length) }}   {{ obj.short_name }}{% if args %}({{ args }}){% endif %}
+{% endfor %}
+{% endif %}
+
+
+   {% if obj.bases %}
+   {% if "show-inheritance" in autoapi_options %}
+   Bases: {% for base in obj.bases %}{{ base|link_objs }}{% if not loop.last %}, {% endif %}{% endfor %}
+   {% endif %}
+
+
+   {% if "show-inheritance-diagram" in autoapi_options and obj.bases != ["object"] %}
+   .. autoapi-inheritance-diagram:: {{ obj.obj["full_name"] }}
+      :parts: 1
+      {% if "private-members" in autoapi_options %}
+      :private-bases:
+      {% endif %}
+
+   {% endif %}
+   {% endif %}
+   {% if "inherited-members" in autoapi_options %}
```

Comment:
can this file be documented/commented in some meaningful way?  Was it copied and modified from something existing?  Maybe inlining comments with examples of what any of the complex secitons or non-standard parts are intended to render?  this may be of little value so feel free to disregard.

Reply:
All the templates are slightly modified versions from the upstream projects. I could at least link to the repository and commit I got them from though.

 - [ ] Add upstream links to templates.